pkg/api/validate.go

```diff
@@ -24,6 +24,10 @@ var (
 		`$`)
 
 	rxAgentPoolProfileName = regexp.MustCompile(`(?i)^[a-z0-9]{1,12}$`)
+
+	// This regexp is to guard against InvalidDomainNameLabel for hostname validation
+	rxCloudDomainLabel = regexp.MustCompile(`(?i)^` +
```

Comment:
@kwoodson looks like we need to remove `(?i)` here - see https://github.com/openshift/openshift-azure/issues/553
@sozercan fyi